robot/client/client.go

```diff
@@ -494,12 +495,18 @@ func (rc *RobotClient) ResourceByName(name resource.Name) (interface{}, error) {
 		return client, nil
 	}
 
-	resourceClient, err := rc.createClient(name)
-	if err != nil {
-		return nil, err
+	// before adding a new resource, make sure this name exists and is known
```

Comment:
NIT: the comment just above (one final check) isn't quite true anymore

Reply:
ahh good catch! updating now :)